midealocal/cli.py

```diff
@@ -14,116 +14,202 @@
 import platformdirs
 from colorlog import ColoredFormatter
 
-from midealocal.cloud import clouds, get_midea_cloud
+from midealocal.cloud import MideaCloud, clouds, get_midea_cloud
 from midealocal.device import ProtocolVersion
 from midealocal.devices import device_selector
 from midealocal.discover import discover
 from midealocal.exceptions import ElementMissing
 from midealocal.version import __version__
 
 _LOGGER = logging.getLogger("cli")
-
-
-def get_config_file_path(relative: bool = False) -> Path:
-    """Get the config file path."""
-    local_path = Path("midea-local.json")
-    if relative or local_path.exists():
-        return local_path
-    return platformdirs.user_config_path(appname="midea-local").joinpath(
-        "midea-local.json",
-    )
-
-
-async def _get_keys(args: Namespace, device_id: int) -> dict[int, dict[str, Any]]:
-    if not args.cloud_name or not args.username or not args.password:
-        raise ElementMissing("Missing required parameters for cloud request.")
-    async with aiohttp.ClientSession() as session:
-        cloud = get_midea_cloud(
-            cloud_name=args.cloud_name,
-            session=session,
-            account=args.username,
-            password=args.password,
+LOG_FORMAT = (
+    "%(asctime)s.%(msecs)03d %(levelname)s (%(threadName)s) [%(name)s] %(message)s"
+)
+
+
+class MideaCLI:
+    """Midea CLI."""
+
+    session: aiohttp.ClientSession
+    namespace: Namespace
+
+    async def _get_cloud(self) -> MideaCloud:
+        """Get cloud instance."""
+        if not hasattr(self, "session"):
```

Comment:
Check session after checking if params are provided. Otherwise you initiate a session without a real need.
On top of that with current design you should handle the session close before raising.

Reply:
Good catch.